### ISSUE TITLE
Throwing FileNotFoundException crashes on iOS

### ISSUE THREAD

**Issue by jperedadnr** [oracle/graal#2189]
**Describe GraalVM and your environment :**
 - GraalVM version or commit id if built from source: **[custom build 20.1.0]**
 - CE or EE: **[CE]**
 - Build Time or run time failure: **[run-time]**
 - JDK version: **[JDK11]**
 - Native compiler information:

```
Apple clang version 11.0.0 (clang-1100.0.33.16)
Target: x86_64-apple-darwin19.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

 - OS and OS Version: **[macOS Catalina 10.15.2]**
 - Architecture: **[ARM64-iOS]**
 - The output of `java -Xinternalversion`: 
```
OpenJDK 64-Bit Server VM (11.0.2+7) for bsd-amd64 JRE (11.0.2+7), built on Dec 12 2018 04:24:53 by "mach5one" with gcc 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)
```

**Have you verified this issue still happens when using the latest snapshot?**
This issue happens with a custom build from master
 
**Describe the issue**
Running a [HelloWorld sample](https://github.com/gluonhq/client-samples/tree/master/Maven/HelloWorld), modified with the snippet below, on iOS causes the following crash, if a `FileNotFoundException` occurs because the expected file doesn't exist, with the following stacktrace:

```
 Process 23801 stopped
 * thread #9, stop reason = EXC_BAD_ACCESS (code=1, address=0x6)
     frame #0: 0x00000001048daf04 helloWorld`JNIJavaCallWrappers__jniInvoke__VA__LIST_003aLjava__io__FileNotFoundException__2__0002e__0003cinit__0003e__00028Ljava__lang__String__2Ljava__lang__String__2__00029V__d01b5d4b7a7694fdaee18008b584fb6dbc616a97 + 228
 helloWorld`JNIJavaCallWrappers__jniInvoke__VA__LIST_003aLjava__io__FileNotFoundException__2__0002e__0003cinit__0003e__00028Ljava__lang__String__2Ljava__lang__String__2__00029V__d01b5d4b7a7694fdaee18008b584fb6dbc616a97:
 ->  0x1048daf04 <+228>: ldr    x1, [x8], #0x8
     0x1048daf08 <+232>: str    x8, [x19]
     0x1048daf0c <+236>: ldur   x8, [x29, #-0x18]
     0x1048daf10 <+240>: str    x0, [sp, #0x20]
 Target 0: (helloWorld) stopped.
 * thread #9, stop reason = EXC_BAD_ACCESS (code=1, address=0x6)
   * frame #0: 0x00000001048daf04 helloWorld`JNIJavaCallWrappers__jniInvoke__VA__LIST_003aLjava__io__FileNotFoundException__2__0002e__0003cinit__0003e__00028Ljava__lang__String__2Ljava__lang__String__2__00029V__d01b5d4b7a7694fdaee18008b584fb6dbc616a97 + 228
     frame #1: 0x0000000104cafbc4 helloWorld`JNU_NewObjectByName + 156
     frame #2: 0x0000000104caeb3c helloWorld`throwFileNotFoundException + 104
     frame #3: 0x0000000104caeebc helloWorld`fileOpen + 376
     frame #4: 0x0000000104336f0c helloWorld`__llvm_jni_wrapper_f_isVoidi64i64i64f + 32
     frame #5: 0x000000010450a9b4 helloWorld`FileInputStream_open0_8b1b5fc3043b8ae0dd0d8b68b3a5a224719f9dca + 200
     frame #6: 0x00000001044ffaa4 helloWorld`FileInputStream_constructor_aac741e2ef07c171301eb35c0deea3293fe4d747 + 372
     frame #7: 0x0000000104503334 helloWorld`FileInputStream_constructor_c26cfb3083610a614cb5843727b1293a2e050fbe + 140
     frame #8: 0x000000010484c210 helloWorld`HelloWorld_main_6ed547d50a30bc97dec85c6ce61dc044222e770a + 160
     frame #9: 0x0000000104914aa4 helloWorld`IsolateEnterStub_JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b_a61fe6c26e84dd4037e4629852b5488bfcc16e7e + 316
     frame #10: 0x000000010423836c helloWorld`startGVM + 40
     frame #11: 0x00000001042380b8 helloWorld`-[AppDelegate startVM:] + 40
     frame #12: 0x00000001947070a4 Foundation`__NSThread__start__ + 848
     frame #13: 0x0000000194039d8c libsystem_pthread.dylib`_pthread_start + 156
```

**Describe the full native-image command**
<details><summary>Capture full native-image command by running with the `--verbose` flag e.g.:</summary>

```
 native-image --report-unsupported-elements-at-runtime -Djdk.internal.lambda.eagerlyInitialize=false -H:+ExitAfterRelocatableImageWrite -H:TempDirectory=/Users/JosePereda/Gluon/Gluon-OSS/Client/Forks/jpereda-client-samples/Maven/HelloWorld/target/client/arm64-ios/gvm/tmp -H:+SharedLibrary -H:+AddAllCharsets -H:EnableURLProtocols=http,https -H:ReflectionConfigurationFiles=/tmp/HelloWorld/target/client/arm64-ios/gvm/reflectionconfig-arm64-ios.json -H:JNIConfigurationFiles=/tmp/HelloWorld/target/client/arm64-ios/gvm/jniconfig-arm64-ios.json -H:ResourceConfigurationFiles=/tmp/HelloWorld/target/client/arm64-ios/gvm/resourceconfig-arm64-ios.json -H:CompilerBackend=llvm -H:-SpawnIsolates -Dllvm.bin.dir=/tmp/graalvm-svm-darwin-20.1.0-ea+24/lib/llvm/bin -H:+UseCAPCache -H:CAPCacheDir=/tmp/HelloWorld/target/client/arm64-ios/gvm/capcache -Dsvm.targetName=iOS -Dsvm.targetArch=arm64 -Dsvm.platform=org.graalvm.nativeimage.Platform$DARWIN_AARCH64 -cp /tmp/HelloWorld/target/classes hello.HelloWorld

```
</details>

```
Build on Server(pid: 33841, port: 59405)
[hello.helloworld:33841]    classlist:     250.02 ms,  2.07 GB
[hello.helloworld:33841]        (cap):      77.93 ms,  2.07 GB
[hello.helloworld:33841]        setup:     477.68 ms,  2.07 GB
[hello.helloworld:33841]   (typeflow):  13,636.38 ms,  2.92 GB
[hello.helloworld:33841]    (objects):  12,115.29 ms,  2.92 GB
[hello.helloworld:33841]   (features):     879.67 ms,  2.92 GB
[hello.helloworld:33841]     analysis:  27,255.75 ms,  2.92 GB
[hello.helloworld:33841]     (clinit):     446.53 ms,  2.92 GB
[hello.helloworld:33841]     universe:     961.52 ms,  2.92 GB
[hello.helloworld:33841]      (parse):   1,453.13 ms,  2.89 GB
[hello.helloworld:33841]     (inline):   1,867.66 ms,  2.64 GB
[hello.helloworld:33841]    (compile):  18,111.99 ms,  2.93 GB
[hello.helloworld:33841]    (bitcode):   1,426.13 ms,  2.93 GB
[hello.helloworld:33841]    (prelink):   4,438.60 ms,  2.93 GB
[hello.helloworld:33841]       (llvm):  41,748.94 ms,  2.93 GB
[hello.helloworld:33841]   (postlink):   2,501.81 ms,  2.93 GB
[hello.helloworld:33841]      compile:  71,753.71 ms,  2.93 GB
[hello.helloworld:33841]        image:   4,686.64 ms,  2.97 GB
[hello.helloworld:33841]        write:     916.59 ms,  2.97 GB
[hello.helloworld:33841]      [total]: 106,430.14 ms,  2.97 GB

```

**Code snippet or code repository that reproduces the issue**
```
public static void main(String[] args) throws Exception{
        try {
            new FileInputStream("nonexistent.css");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
```

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. Step one [Download GraalVM build for [iOS](https://download2.gluonhq.com/substrate/graalvm/graalvm-svm-darwin-20.1.0-ea+24.zip), extract and set `$GRAALVM_HOME`]
2. Step two [Clone the [sample](https://github.com/gluonhq/client-samples), modify with above snippet, and enable iOS target in pom]
3. Step three [`mvn clean client:build`]
4. Step four [deploy to an iOS device, following [documentation](https://docs.gluonhq.com/client/#_ios_deployment): `mvn client:run`]

**Expected behavior**
The `FileNotFoundException` should be thrown, as happens when running the same sample on MacOS:

```
 java.io.FileNotFoundException: nonexistent.css (No such file or directory)
    at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST:Ljava_io_FileNotFoundException_2_0002e_0003cinit_0003e_00028Ljava_lang_String_2Ljava_lang_String_2_00029V(JNIJavaCallWrappers.java:0)
    at java.io.FileInputStream.open0(FileInputStream.java)
    at java.io.FileInputStream.open(FileInputStream.java:219)
...
```

**Additional context**
Possibly related to #2090

**Comment by johanvos** [comment]
commit id: 1e2cf8573de982d009183642b2fb04ae6bcfdda7

**Comment by loicottet** [comment]
Closed by [edec559b](https://github.com/oracle/graal/commit/edec559b198885727b1a81ef3e50b3b76a834d68)
